packages/@aws-cdk/toolkit-lib/lib/api/diff/diff-formatter.ts

```diff
@@ -329,6 +363,36 @@ export class DiffFormatter {
     const formattedDiff = stream.toString();
     return { formattedDiff, permissionChangeType: this.permissionType() };
   }
+
+  public formatStackDrift(options: FormatStackDriftOptions): FormatStackDriftOutput {
+    const stream = new StringWriteStream();
+    let driftCount = 0;
+
+    if (!this.driftResults?.StackResourceDrifts) {
+      return { formattedDrift: '', numResourcesWithDrift: 0 };
+    }
```

Comment:
These seems incorrect. It's not `0` resources with drift - it's unkown.

Reply:
Good point. The chunk is supposed to catch if the drift results are empty when the method is called, which right now doesn't seem like it will happen with my implementation, but I agree it would return 0 when really drift detection hasn't run. Would it be better to return a more descriptive message, maybe with `numResourcesWithDrift: -1`?

Reply:
Yeah, that kind of highlights a weakness in the approach of introducing an optional prop into the formatter class constructor, to later maybe call the format function.

As far as I can tell, there is no reason this needs to be part of `DiffFormatter`. So let's just make it it's own file and api: `drift/formatter.ts` or whatever. You can probably also simplify how the class handles the formatting since the detour via `stream` is just an artifact of legacy code. Instead pushing onto an array and joining with newlines will do the same.